File: 88. Merge Sorted Array.java
/*This is my first ever leetcode problem *-* 
 * As I was a beginner with no big idea I just proceeded with the simple thought of adding the two arrays into one array and
 * then sorting it. Which is not optimal at all. Thus I might redo another submission for this problem soon.
 */
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int j;
        int i=0;
        int k=0;
        boolean status = true;
        int [] result = new int[m+n];
        for(i=0;i<m;i++){
            result[k]=nums1[i];
            k++;
           // System.out.println(result[k]);
        }
        for(j=0;j<n;j++){
            result[k]=nums2[j];
            k++;
          //  System.out.println(result[k]);
        }
        while(status){
            status=false;
            for(k=1;k<m+n;k++){
                if(result[k-1]>result[k]){
                    j=result[k-1];
                    result[k-1]=result[k];
                    result[k]=j;
                    status=true;
                }
            }
        }
        for(k=0;k<m+n;k++){
                System.out.println(result[k]);
                nums1[k]=result[k];
                }

}}

File: 88. Merge Sorted Array.py
from collections import List
"""
As this is my first ever problem on leetcode, I did it the first time using java and the solution was not optimal and it had a bad 
runtime and space usage. But now, as I am 128 problems in, I went back and did it again. 
The solution is really simple, just merging the two tables as if it was a mergeSort last step
"""
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        temp = nums1.copy()
        temp = temp[:m]
        i = 0
        j = 0
        k = 0
        while k<m+n:
            if i>=m:
                nums1[k]=nums2[j]
                j+=1
            elif j<n and i<m and nums2[j] <= temp[i]:
                nums1[k]=nums2[j]
                j+=1
            else:
                nums1[k]=temp[i]
                i+=1
            k+=1